tests/ui/unit_arg.stderr

```diff
@@ -1,79 +1,211 @@
 error: passing a unit value to a function
-  --> $DIR/unit_arg.rs:24:9
+  --> $DIR/unit_arg.rs:23:5
    |
 LL |     foo({});
-   |         ^^
+   |     ^^^^^^^
    |
    = note: `-D clippy::unit-arg` implied by `-D warnings`
-help: if you intended to pass a unit value, use a unit literal instead
+help: move the expression in front of the call...
+   |
+LL |     {};
```

Comment:
Hmm, so the `{}` is an expression (a block) that returns `()`? Shouldn't the lint still provide the old suggestion here, instead of the two new ones? I don't think there's any point in moving an empty block expression out of the function args?

Reply:
This would be handled by the `no_effect` lint afterwards I think. But special casing empty blocks would definitely make sense here. I'll add this to the suggestion generation.